katas/content/superposition/wstate_power_of_two/Verification.qs

```diff
@@ -0,0 +1,84 @@
+namespace Kata.Verification {
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Katas;
+
+    operation WState_PowerOfTwo_Reference (qs : Qubit[]) : Unit is Adj+Ctl {
+        let N = Length(qs);
+
+        if N == 1 {
+            X(qs[0]);
+        } else {
+            let K = N / 2;
+            use anc = Qubit();
+            H(anc);
+
+            ApplyControlledOnInt(0, WState_PowerOfTwo_Reference, [anc], qs[0 .. K - 1]);
+            ApplyControlledOnInt(1, WState_PowerOfTwo_Reference, [anc], qs[K .. N - 1]);
+
+            for i in K .. N - 1 {
+                CNOT(qs[i], anc);
+            }
+        }
+    }
+
+    function FindFirstDiff (bits1 : Bool[], bits2 : Bool[]) : Int {
+        for i in 0 .. Length(bits1) - 1 {
+            if bits1[i] != bits2[i] {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    operation TwoBitstringSuperposition_Reference (qs : Qubit[], bits1 : Bool[], bits2 : Bool[]) : Unit is Adj + Ctl {
+        let firstDiff = FindFirstDiff(bits1, bits2);
+
+        H(qs[firstDiff]);
+
+        for i in 0 .. Length(qs) - 1 {
+            if bits1[i] == bits2[i] {
+                if bits1[i] {
+                    X(qs[i]);
+                }
+            } else {
+                if i > firstDiff {
+                    CNOT(qs[firstDiff], qs[i]);
+                    if bits1[i] != bits1[firstDiff] {
+                        X(qs[i]);
+                    }
+                }
+            }
```

Comment:
This approach works nicely for N=1 and N=2, but if the solution happens to be correct for these two but incorrect for one of the larger Ns, instead of a nice readable message we get a crash because of the use of an auxiliary qubit in the reference solution (if the solution is incorrect, applying adjoint of the reference solution ends with this aux qubit in a non-zero state and an error about that). 

This is very confusing, so I think we need to sneak a peek at the next task and use the reference solution from it that uses arbitrary rotations but not extra qubits - this way we always get readable results.

```suggestion
    open Microsoft.Quantum.Math;

    operation WState_PowerOfTwo_Reference (qs : Qubit[]) : Unit is Adj + Ctl {
        let N = Length(qs);
        Ry(2.0 * ArcSin(Sqrt(1.0/IntAsDouble(N))), qs[0]);
        for i in 1 .. N - 1 {
            ApplyControlledOnInt(0, Ry(2.0 * ArcSin(Sqrt(1.0/IntAsDouble(N - i))), _), qs[0 .. i-1], qs[i]);
```